src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,39 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
 public class SalaryInfo {
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        String[] salaries = new String[names.length];
+        int index = 0;
+        int income = 0;
+        final DateTimeFormatter formatter = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+        LocalDate startDate = LocalDate.parse(dateFrom, formatter);
+        LocalDate finishDate = LocalDate.parse(dateTo, formatter);
+        for (String name : names) {
+            for (int i = 0; i < data.length; i++) {
+                String[] personalData = data[i].split(" ");
+                LocalDate currentDate = LocalDate.parse(personalData[0], formatter);
+                if (name.equals(personalData[1]) && currentDate.compareTo(startDate) >= 0
+                        && currentDate.compareTo(finishDate) <= 0) {
```

Comment:
use isBefore, isAfter methods of LocalDate for comparison
extract indexes to constants 0 1  2 3